core/memory_tracker/README.md

```diff
@@ -0,0 +1,64 @@
+# Track Coherent Memory
+
+## Problem and Solution
+Coherent memories are implicitly synchronized which means there is no *flush*
+or *invalid* command to explicitly *write* or *read* to/from those memories.
+Henceforth we cannot rely on intercepting the *flush* and *invalid* commands to
+track the modifications for coherent memory.
+
+The solution is to utilize segmentation fault signal handler to tell to which
+memory page has been written and then read the data of that page to track the
+data in coherent memory.
+
+## Goal
+The goal is to implement a memory tracker that:
+* Tracks the memory pages that have been written to.
+* Transparent to the application under debugging.
+ * Assumes zero modification on the application under debugging
+ * For real segmentation faults, the system handler or, if any, the handler
```

Comment:
For any memory that is not tracked, the previously installed handler should be called.